.github/workflows/main.yml

```diff
@@ -30,3 +30,36 @@ jobs:
         run: python -m pip install -U tox
       - name: Run lint
         run: tox -elint
+  build:
+    runs-on: ubuntu-latest  # windows-latest | macos-latest
+    name: Test changed-files
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # OR "2" -> To retrieve the preceding commit.
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v24
+        with:
+          files: |
+            red_queen/games/applications/*.py
+            red_queen/games/mapping/*.py
+          files_ignore: |
+            red_queen/games/applications/__init__.py
+            red_queen/games/mapping/__init__.py
+      - name: Install Deps
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest-sugar
+          pip install .
+      - name: Run every new benchmark
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_and_modified_files }}; do
+            pytest ${file} -n 2 --verbose --tb=no --store
+            python -m report.console_tables --storage results/0001_bench.json
+            rm results/0001_bench.json
```

Comment:
We don't really need to remove this, the VM gets shutdown at the end of the run so we don't need to worry about dangling files.

Reply:
I am aware of that. I'm just removing it in case more than one file is to be tested. In such case, since there's no indexing in these loops, I just remove the file. So when the next benchmark runs it generates the outputs to 0001_bench again.